### ISSUE TITLE
Exclude NSX segments

### ISSUE THREAD

**Issue by a084ed22** [bl4ko/netbox-ssot#85]
Hello,

I've tested this against a vCenter bound to NSX and with some NSX-managed overlay segments. They seem to be read as regular portgroups/VLANs, however attempts to create or modify them on Netbox fails as their VLAN ID is unset.

```
rest.go:132         DEBUG   (vcenter-test): Creating objects.Vlan with path /api/ipam/vlans/ with data: Vlan{ID: 0, Name: NSX-TEST, Vid: 0, Status: active}
main.go:101         ERROR   (vcenter-test): unexpected status code: 400: {"vid":["This field is required."]}
main.go:120         INFO    (main): Skipping removing orphaned objects...
main.go:130         INFO    (main): ⚠ syncing of source cor-vvc-10-p-ar failed
```

I'm not sure what additional information I can provide.

**Comment by bl4ko** [comment]
Hello, thanks for opening the issue. Currently, there isn't any specific logic implemented yet to handle NSX segments. I am also not particularly familiar with this feature, but I will take a look into it. I see that the problem in the above example is that the VID is set to 0, which is then not sent to API because of the [omitempty](https://github.com/bl4ko/netbox-ssot/blob/2e3b79d2ee63cfcfa3bd05adf09c1c4a12581cae/internal/netbox/objects/ipam.go#L99) (which could be the issue).

In the meantime, if you are maybe familiar with go, the code for handling networks can be found [here](https://github.com/bl4ko/netbox-ssot/blob/2e3b79d2ee63cfcfa3bd05adf09c1c4a12581cae/internal/source/vmware/vmware_init.go#L14) and [here](https://github.com/bl4ko/netbox-ssot/blob/2e3b79d2ee63cfcfa3bd05adf09c1c4a12581cae/internal/source/vmware/vmware_sync.go#L18). Also if you could provide longer debug output, or vsphere network api values that would be great.

**Comment by a084ed22** [comment]
I'll try to produce some, unfortunately I have not had sufficient time to dedicate to this.

**Comment by bl4ko** [comment]
This has been solved in #129. Reopen if needed.
